src/main/scala/scala/tools/refactoring/implementations/MarkOccurrences.scala

```diff
@@ -59,7 +61,61 @@ trait MarkOccurrences extends common.Selections with analysis.Indexes with commo
     }
   }
 
+  /*
+   * Self references aka `class Foo { self =>` need special treatment, because their representation in typed
+   * ASTs is incomplete. While `self =>` is represented as a synthetic `ValDef`, usages of the named reference
+   * can not be distinguished from the `this` keyword, without looking into the source code. This method
+   * takes care of this special case.
+   */
+  private def findOccurrencesForSelfReference(selection: SingleTreeSelection): Option[List[(RangePosition, Tree)]] = {
```

Comment:
I already wondered how to solve this problem by myself. I guess looking it up in the source code is the best what we can do.